### ISSUE TITLE
Unable to start business network on ubuntu

### ISSUE THREAD

**Issue by nasht00** [hyperledger/composer#4478]
I have written the following script, which is pretty much step by step the developer tutorial:

```
export FABRIC_HOME=~/fabric-dev-servers
export FABRIC_VERSION=hlfv12
(cd $FABRIC_HOME && ./teardownFabric.sh)
docker kill $(docker ps -q)
docker rm $(docker ps -aq)
docker rmi $(docker images dev-* -q)
rm -rf ~/.composer/cards/*
rm -rf ~/.composer/client-data/*
(cd $FABRIC_HOME && ./startFabric.sh)
(cd $FABRIC_HOME && ./createPeerAdminCard.sh)
composer archive create -t dir -n .
composer network install --card PeerAdmin@hlfv1 --archiveFile tutorial-network@0.0.1.bna
composer network start --networkName tutorial-network --networkVersion 0.0.1 --networkAdmin admin --networkAdminEnrollSecret adminpw --card PeerAdmin@hlfv1 --file networkadmin.card
composer card import --file networkadmin.card
composer network ping --card admin@tutorial-network
composer-rest-server -c admin@tutorial-network -n never -u true -w true
```

On my mac, it all works fine all the way to the REST server.
On my Ubuntu VM however, it fails at `composer network start` with the following errors:

```
Starting business network tutorial-network at version 0.0.1

Processing these Network Admins: 
	userName: admin

✖ Starting business network definition. This may take a minute...
Error: Error trying to start business network. Error: No valid responses from any peers.
Response from attempted peer comms was an error: Error: failed to execute transaction 67a975709935effd35dad8295f0b55b33ad40f52772ec44d2b220edd6b1f6ace: error starting container: error starting container: Failed to generate platform-specific docker build: Error returned from build: 1 "npm WARN deprecated hoek@2.16.3: The major version is no longer supported. Please update to 4.x or newer

> x509@0.3.3 install /chaincode/output/node_modules/x509
> node-gyp rebuild

gyp ERR! configure error 
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at failNoPython (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:401:14)
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:330:11
gyp ERR! stack     at F (/usr/local/lib/node_modules/npm/node_modules/which/which.js:69:16)
gyp ERR! stack     at E (/usr/local/lib/node_modules/npm/node_modules/which/which.js:81:29)
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/which/which.js:90:16
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/which/node_modules/isexe/index.js:44:5
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/which/node_modules/isexe/access.js:8:5
gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:123:15)
gyp ERR! System Linux 4.4.0-135-generic
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /chaincode/output/node_modules/x509
gyp ERR! node -v v6.7.0
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok 
tutorial-network@0.0.1 /chaincode/output
`-- composer-common@0.20.2 
  +-- acorn@5.1.2 
  +-- axios@0.17.1 
  | +-- follow-redirects@1.5.9 
  | | `-- debug@3.1.0 
  | `-- is-buffer@1.1.6 
  +-- comment-parser@0.4.0 
  | `-- readable-stream@2.3.6 
  |   +-- core-util-is@1.0.2 
  |   +-- inherits@2.0.3 
  |   +-- process-nextick-args@2.0.0 
  |   +-- safe-buffer@5.1.2 
  |   +-- string_decoder@1.1.1 
  |   `-- util-deprecate@1.0.2 
  +-- config@1.24.0 
  | `-- json5@0.4.0 
  +-- debug@2.6.9 
  | `-- ms@2.0.0 
  +-- doctrine@2.0.0 
  | +-- esutils@2.0.2 
  | `-- isarray@1.0.0 
  +-- fs-extra@1.0.0 
  | +-- graceful-fs@4.1.11 
  | +-- jsonfile@2.4.0 
  | `-- klaw@1.3.1 
  +-- js-yaml@3.10.0 
  | +-- argparse@1.0.10 
  | `-- esprima@4.0.1 
  +-- jsrsasign@8.0.3 
  +-- jszip@3.1.3 
  | +-- core-js@2.3.0 
  | +-- es6-promise@3.0.2 
  | +-- lie@3.1.1 
  | | `-- immediate@3.0.6 
  | +-- pako@1.0.6 
  | `-- readable-stream@2.0.6 
  |   +-- process-nextick-args@1.0.7 
  |   `-- string_decoder@0.10.31 
  +-- lodash.clonedeep@4.5.0 
  +-- lodash.padstart@4.5.0 
  | +-- lodash._baseslice@4.0.0 
  | +-- lodash._basetostring@4.12.0 
  | `-- lodash.tostring@4.1.4 
  +-- lorem-ipsum@1.0.4 
  | `-- optimist@0.3.7 
  |   `-- wordwrap@0.0.3 
  +-- minimatch@3.0.3 
  | `-- brace-expansion@1.1.11 
  |   +-- balanced-match@1.0.0 
  |   `-- concat-map@0.0.1 
  +-- mkdirp@0.5.1 
  | `-- minimist@0.0.8 
  +-- npm-paths@0.1.3 
  | +-- global-modules@0.2.3 
  | | +-- global-prefix@0.1.5 
  | | | +-- homedir-polyfill@1.0.1 
  | | | | `-- parse-passwd@1.0.0 
  | | | +-- ini@1.3.5 
  | | | +-- is-windows@0.2.0 
  | | | `-- which@1.3.1 
  | | |   `-- isexe@2.0.0 
  | | `-- is-windows@0.2.0 
  | `-- is-windows@0.1.1 
  +-- semver@5.3.0 
  +-- sprintf-js@1.0.3 
  +-- thenify@3.2.1 
  | `-- any-promise@1.3.0 
  +-- thenify-all@1.6.0 
  +-- urijs@1.19.1 
  +-- uuid@3.0.1 
  `-- winston@2.3.1 
    +-- async@1.0.0 
    +-- colors@1.0.3 
    +-- cycle@1.0.3 
    +-- eyes@0.1.8 
    +-- isstream@0.1.2 
    `-- stack-trace@0.0.10 

npm WARN tutorial-network@0.0.1 No repository field.
npm WARN tutorial-network@0.0.1 license should be a valid SPDX license expression
npm ERR! Linux 4.4.0-135-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "--production"
npm ERR! node v6.7.0
npm ERR! npm  v3.10.3
npm ERR! code ELIFECYCLE

npm ERR! x509@0.3.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the x509@0.3.3 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the x509 package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs x509
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls x509
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /chaincode/output/npm-debug.log
"
Command failed
```

Some information about the Ubuntu VM:

- Ubuntu 16.04.5 LTS (GNU/Linux 4.4.0-135-generic x86_64)
- Docker version 18.06.1-ce, build e68fc7a
- docker-compose version 1.11.2, build dfed245
- node -v v8.12.0
- npm -v 6.4.1
- git version 2.7.4
- Python 2.7.12

I see several errors, not sure which is the root one.
One of them mentions python. I am able to type `python -V` successfully so I'm not sure what it is complaining about. In any case I've tried adding this line to the top of the script: `export PYTHON=/usr/bin/python`, but it didn't help.

UPDATE: I ran a `docker logs` on the peer, and see the same error there about python and node-gyp. So that explains why my local python binary does not help. The error is within the docker container...

**Comment by davidkel** [comment]
Doesn't look like you are running an appropriate level of fabric
```
npm ERR! node v6.7.0
npm ERR! npm  v3.10.3
```
I would suggest you delete all your containers and docker images and start again. It looks like you care using some very old chaincode builder images.

**Comment by nasht00** [comment]
Strangely I installed everything from scratch less than a week ago. Let me try again.

**Comment by nasht00** [comment]
I have modified my setup script to completely wipe off previous install and download from scratch:

```
export FABRIC_HOME=~/fabric-dev-servers
export FABRIC_VERSION=hlfv12
npm uninstall -g composer-cli composer-rest-server generator-hyperledger-composer yo
npm install -g composer-cli@0.20 composer-rest-server@0.20 generator-hyperledger-composer@0.20 yo
rm -Rf $FABRIC_HOME
mkdir $FABRIC_HOME
(cd $FABRIC_HOME && curl -O https://raw.githubusercontent.com/hyperledger/composer-tools/master/packages/fabric-dev-servers/fabric-dev-servers.tar.gz)
(cd $FABRIC_HOME && tar -xvf fabric-dev-servers.tar.gz)
(cd $FABRIC_HOME && ./teardownFabric.sh)
docker kill $(docker ps -q)
docker rm $(docker ps -aq)
docker rmi $(docker images dev-* -q)
rm -rf ~/.composer/cards/*
rm -rf ~/.composer/client-data/*
(cd $FABRIC_HOME && ./downloadFabric.sh)
(cd $FABRIC_HOME && ./startFabric.sh)
(cd $FABRIC_HOME && ./createPeerAdminCard.sh)
composer archive create -t dir -n .
composer network install --card PeerAdmin@hlfv1 --archiveFile volvo-network@0.0.1.bna
composer network start --networkName volvo-network --networkVersion 0.0.1 --networkAdmin admin --networkAdminEnrollSecret adminpw --card PeerAdmin@hlfv1 --file networkadmin.card
composer card import --file networkadmin.card
composer network ping --card admin@volvo-network
composer-rest-server -c admin@volvo-network -n never -u true -w true
```

Yet I get the exact same error.

Also, if it helps, the results of `docker ps`:

```
CONTAINER ID        IMAGE                               COMMAND                  CREATED             STATUS              PORTS                                            NAMES
a8a82f50f919        hyperledger/fabric-peer:1.2.1       "peer node start"        4 minutes ago       Up 4 minutes        0.0.0.0:7051->7051/tcp, 0.0.0.0:7053->7053/tcp   peer0.org1.example.com
bf57ffe93166        hyperledger/fabric-ca:1.2.1         "sh -c 'fabric-ca-se…"   4 minutes ago       Up 4 minutes        0.0.0.0:7054->7054/tcp                           ca.org1.example.com
9a9bc435b292        hyperledger/fabric-couchdb:0.4.10   "tini -- /docker-ent…"   4 minutes ago       Up 4 minutes        4369/tcp, 9100/tcp, 0.0.0.0:5984->5984/tcp       couchdb
08cbea7eddea        hyperledger/fabric-orderer:1.2.1    "orderer"                4 minutes ago       Up 4 minutes        0.0.0.0:7050->7050/tcp                           orderer.example.com
```

**Comment by nasht00** [comment]
Ah something happened, instead of running `(docker rmi $(docker images dev-* -q)` (which I noticed did not do anything), I ran `docker rmi $(docker images -q)`.

This time it did the trick!

**Comment by davidkel** [comment]
the second command will have deleted the ccenv image which is the one I think was old and not valid.

**Comment by davidkel** [comment]
Closing as resolved.
